### PR TITLE
[CIFuzz] Refactor in preparation for filestore

### PR DESCRIPTION
**Pull request by jonathanmetzman**
Instead of landing #5299 I'm breaking it up into smaller, more reviewable PRs.
This PR does some of the non-functional refactoring that was initially done by that PR.
The main changes are:
1. Moving the _create_config and create_config functions from build_fuzzers_test.py and run_fuzzers_test.py into test_helpers.py (which is now part of cifuzz instead of infra) and share code between them.
2. Rename artifacts_dir to crashes_dir in run_fuzzers.py. "artifacts" is ambiguous.
3. Make some small changes to pytest.ini to improve debugging.
4. Fix the logging method used in retry (this is the only functional change).

### CODE REVIEW

File: infra/pytest.ini [hunk 1]
```diff
@@ -1,2 +1,3 @@
 [pytest]
-python_files = *_test.py
\ No newline at end of file
+python_files = *_test.py
+log_cli = true
```
Comment (inferno-chromium) [inline infra/pytest.ini:3]: nit: missing new line

**Review by inferno-chromium — APPROVED** [review]

File: infra/cifuzz/actions/run_fuzzers/action.yml [hunk 2]
```diff
@@ -30,6 +30,9 @@ inputs:
       "batch" is in alpha and should not be used in production.
     required: false
     default: 'ci'
+  github-token:
```
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply infra/cifuzz/actions/run_fuzzers/action.yml:33]: Users are supposed to specify secrets.GITHUB_TOKEN in their cifuzz.yml.

[the remark this answers is not in the document]
Comment (inferno-chromium) [reply infra/cifuzz/actions/run_fuzzers/action.yml:33]: Can you add a comment to tell people to use secrets.GITHUB_TOKEN explicitly, dont want people to mistake for adding their token directly. also good to add link to https://docs.github.com/en/actions/reference/authentication-in-a-workflow in comment

[the remark this answers is not in the document]
Comment (oliverchang) [reply infra/cifuzz/actions/run_fuzzers/action.yml:33]: We'll also want to have documentation to tell users to restrict the scope of that token by setting https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions

**Review by oliverchang — APPROVED** [review]

File: infra/cifuzz/actions/run_fuzzers/action.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply infra/cifuzz/actions/run_fuzzers/action.yml:33]: Done.

**Pull request merged**